### PR TITLE
rfct: block container

### PR DESCRIPTION
**Pull request by shanberg**
Goal: move ~magic~ complex behavior from block `Container` component to block `core` cljs

Major
- Context menu lives in `core.cljs`, not `Container.tsx`
- Context menu content is shared between block and block anchor
- Block `show-edit?` refactored to `is-hovered-not-child?`, and only applies to the currently-hovered block, not also all ancestors.

Minor
- Cleanup in block anchor, toggle, content, and container components
- Fixes a cause of block toggles appearing when not wanted
- Better alert style when blocks fail to render

### CODE REVIEW

File: src/cljs/athens/views/blocks/core.cljs [hunk 1]
```diff
@@ -401,86 +457,58 @@
...
+
+         [:> Container {:isActive          has-menu-open?
+                        :isDragging        (and @dragging? (not @selected?))
+                        :isHoveredNotChild @is-hovered-not-child?
+                        :isLinkedRef       (and (false? initial-open) (= uid linked-ref-uid))
+                        :isOpen            open
+                        :isSelected        @selected?
+                        :hasChildren       (seq children)
+                        :uid               uid
+                        :ref               container-ref
```
Comment (shanberg) [inline src/cljs/athens/views/blocks/core.cljs:486]: Currently not setting the ref here properly, not sure why yet

File: src/cljs/athens/views/blocks/core.cljs [hunk 2]
```diff
@@ -401,86 +474,52 @@
...
-                                                       :onClick  #(ctx-menu/handle-click-comment % uid)
-                                                       :icon     (r/as-element [:> ChatBubbleIcon])}])
-                                       (when reactions-enabled?
-                                         [:<>
-                                          [:> MenuDivider]
-                                          [block-reaction/reactions-menu-list uid user-id]])
-                                       (when (and notifications-enabled? (actions/is-block-notification? properties))
-                                         [:> MenuItem {:children "Archive"
-                                                       :icon     (r/as-element [:> ArchiveIcon])
-                                                       :onClick  #(rf/dispatch (actions/update-state-prop uid "athens/notification/is-archived" "true"))}])])
```
Comment (shanberg) [inline src/cljs/athens/views/blocks/core.cljs:447]: Moved to outer function

File: src/cljs/athens/views/blocks/core.cljs [hunk 3]
```diff
@@ -350,17 +425,15 @@
    (let [[block-uid _embed-id]    (-> block :block/uid common-db/uid-and-embed-id)
          {:keys [initial-open
                  parent-uids
-                 linked-ref
-                 linked-ref-uid]} linked-ref-data
+                 linked-ref]} linked-ref-data
          ident                    [:block/uid block-uid]
-         show-edit?               (r/atom false)
-         hide-edit-fn             #(reset! show-edit? false)
-         show-edit-fn             #(reset! show-edit? true)
```
Comment (shanberg) [inline src/cljs/athens/views/blocks/core.cljs:358]: Refactored to `is-hovered-not-child`

File: src/cljs/athens/time_controls.cljs [hunk 4]
```diff
@@ -101,4 +101,4 @@
                        hue (math/floor (/ (* (- 100 percent)
                                              120)
                                           100))]
-                   {:background-color (str "hsl(" hue ",50%,75%)")})))))))
+                   {:backgroundColor (str "hsl(" hue ",50%,75%)")})))))))
```
Comment (shanberg) [inline src/cljs/athens/time_controls.cljs:104]: Renamed to fit `sx` styling API

File: src/cljs/athens/views/blocks/core.cljs [hunk 5]
```diff
@@ -401,86 +474,52 @@
...
+                        :onContextMenu (fn [e]
+                                         (if
+                                           (.includes CONTAINER_CONTEXT_MENU_FILTERED_TAGS (.-tagName (.-target e)))
+                                           (.preventDefault e)
+                                           (.addToContextMenu context-menu
+                                                              #js {:ref container-ref-ref
+                                                                   :event e
+                                                                   :component menu-component
+                                                                   :key "block"})))
+                        :sx        (merge {} (time-controls/block-styles block-o))}
```
Comment (shanberg) [inline src/cljs/athens/views/blocks/core.cljs:522]: Make `Container` dumb again

File: src/js/components/Block/Anchor.tsx [hunk 6]
```diff
@@ -110,78 +79,19 @@ const anchorButtonStyleProps = (isClosedWithChildren: boolean) => {
...
-      {ANCHORS[anchorElement] ? ANCHORS[anchorElement] : anchorElement}
-    </IconButton>
-  </>
-
+    }}
+    {...buttonProps}
+  >
+    {ANCHORS[anchorElement] ? ANCHORS[anchorElement] : anchorElement}
+  </IconButton>
 });
```
Comment (shanberg) [inline src/js/components/Block/Anchor.tsx:95]: Make `Anchor` dumb again

File: src/js/components/Block/Container.tsx [hunk 7]
```diff
@@ -1,63 +1,42 @@
...
-      bg={isMenuOpen ? "background.upper" : undefined}
-      justifyContent="flex-start"
-      flexDirection="column"
-      display="block"
-      background="var(--block-surface-color)"
       opacity={isDragging ? 0.5 : 1}
       data-uid={uid}
       data-childrenuids={childrenUids}
-      sx={{
+      __css={{
```
Comment (shanberg) [inline src/js/components/Block/Container.tsx:32]: `__css` is like `sx` but lower styling priority, so we can more easily override styles when needed. Works when we're building simple/raw components like divs, and not extending defaults like Button.

File: src/js/components/Block/Container.tsx [hunk 8]
```diff
@@ -112,7 +90,9 @@ const _Container = React.forwardRef(({ children, isDragging, isSelected, isOpen,
         ".block-body > .block-toggle:focus": {
           opacity: 1
         },
-        "&.is-hovered-not-child > .block-body > .block-toggle, &:focus-within > .block-body > .block-toggle": { opacity: "1" },
+        "&.is-hovered-not-child > .block-body > .block-toggle, &.is-editing > .block-body > .block-toggle": {
+          opacity: "1"
+        },
```
Comment (shanberg) [inline src/js/components/Block/Container.tsx:95]: Fixes block toggles appearing when any child was focused. Now only the block being edited, or any hovered block should show the toggle.

**Review by filipesilva — APPROVED** [review]

**Pull request merged**